mod.ts

```diff
@@ -330,3 +330,19 @@ export function labelled<T extends Validator>(
     },
   };
 }
+
+/** Creates a validator that matches a custom predicate */
+export function matches<T>(predicate: (val: T) => boolean): Validator<T> {
```

Comment:
I think it might be better to allow having a custom error message here when the predicate is not matched.  It would be optional tho.  Like this:

```suggestion
export function matches<T>(predicate: (val: T) => boolean, customErrorMsg?: string): Validator<T> {
```

And then pass that into `diagnostics.pushError()` if it's not `undefined`.

Reply:
That's a great idea. I'll add that